File: frontend/src/Dashboard/Management_dashboard.cs
* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

.dashboard-container {
  min-height: 100vh;
  background: linear-gradient(135deg, #e8e8e8 0%, #d0d0d0 100%);
  font-family: "Segoe UI", Tahoma, Geneva, Verdana, sans-serif;
}

/* Header */
.dashboard-header {
  background-color: #b00d15;
  color: white;
  padding: 20px 40px;
  text-align: center;
}

.dashboard-header h1 {
  font-size: 2rem;
  font-weight: 600;
  margin: 0;
}

/* Profile Banner */
.profile-banner {
  background-color: #f9e79f;
  padding: 15px 40px;
  display: flex;
  justify-content: space-between;
  align-items: center;
  border-bottom: 1px solid #e0e0e0;
}

.profile-info {
  flex: 1;
}

.profile-info p {
  margin: 0 0 8px 0;
  color: #8b4513;
  font-size: 0.9rem;
}

.progress-bar {
  width: 200px;
  height: 6px;
  background-color: #e0e0e0;
  border-radius: 3px;
  overflow: hidden;
}

.progress-fill {
  width: 25%;
  height: 100%;
  background-color: #28a745;
  border-radius: 3px;
}

.complete-profile-btn {
  background-color: #b00d15;
  color: white;
  border: none;
  padding: 10px 20px;
  border-radius: 5px;
  cursor: pointer;
  font-weight: 500;
  transition: background-color 0.3s;
}

.complete-profile-btn:hover {
  background-color: #8a0a11;
}

/* Dashboard Content */
.dashboard-content {
  display: flex;
  gap: 20px;
  padding: 20px 40px;
  max-width: 1400px;
  margin: 0 auto;
}

.main-section {
  flex: 1;
}

.sidebar {
  width: 300px;
}

/* Dashboard Cards */
.dashboard-card {
  background: white;
  border-radius: 12px;
  padding: 24px;
  box-shadow: 0 2px 10px rgba(0, 0, 0, 0.1);
  margin-bottom: 20px;
}

.dashboard-card h2 {
  font-size: 1.5rem;
  margin-bottom: 20px;
  color: #333;
  text-align: center;
}

/* Dashboard Grid */
.dashboard-grid {
  display: grid;
  grid-template-columns: 1fr 1fr;
  gap: 20px;
  margin-top: 20px;
}

/* Feature Cards */
.feature-card {
  padding: 30px 20px;
  border-radius: 12px;
  text-align: center;
  color: white;
  position: relative;
}

.blue-card {
  background: linear-gradient(135deg, #4a90e2 0%, #357abd 100%);
}

.green-card {
  background: linear-gradient(135deg, #5cb85c 0%, #449d44 100%);
}

.card-icon {
  margin-bottom: 15px;
}

.card-icon svg {
  color: white;
}

.feature-card h3 {
  font-size: 1.2rem;
  margin-bottom: 8px;
  font-weight: 600;
}

.feature-card p {
  font-size: 0.9rem;
  margin-bottom: 5px;
  opacity: 0.9;
}

.next-info,
.current-info {
  font-size: 0.8rem;
  margin-bottom: 15px;
  opacity: 0.8;
}

.action-btn {
  background-color: #b00d15;
  color: white;
  border: none;
  padding: 8px 16px;
  border-radius: 5px;
  cursor: pointer;
  font-size: 0.9rem;
  font-weight: 500;
  transition: background-color 0.3s;
}

.action-btn:hover {
  background-color: #8a0a11;
}

/* Stats Grid */
.stats-grid {
  display: grid;
  grid-template-columns: 1fr 1fr;
  gap: 20px;
  margin-top: 20px;
}

.stat-card {
  background: white;
  border-radius: 12px;
  padding: 24px 24px 30px 24px;
  text-align: center;
  box-shadow: 0 2px 10px rgba(0, 0, 0, 0.1);
}

.stat-icon {
  width: 50px;
  height: 50px;
  border-radius: 50%;
  display: flex;
  align-items: center;
  justify-content: center;
  margin: 0 auto 15px;
}

.green-icon {
  background-color: #d4edda;
  color: #28a745;
}

.purple-icon {
  background-color: #e2d9f3;
  color: #6f42c1;
}

.stat-card h3 {
  font-size: 1rem;
  color: #666;
  margin-bottom: 10px;
  font-weight: 500;
}

.stat-value {
  font-size: 2.5rem;
  font-weight: bold;
  color: #28a745;
  margin-bottom: 5px;
}

.stat-value.purple {
  color: #6f42c1;
}

.stat-label {
  font-size: 0.9rem;
  color: #999;
}

.stat-btn {
  background-color: #b00d15;
  color: white;
  border: none;
  padding: 8px 16px;
  border-radius: 5px;
  cursor: pointer;
  font-size: 0.9rem;
  font-weight: 500;
  margin-top: 15px;
  transition: background-color 0.3s;
}

.stat-btn:hover {
  background-color: #8a0a11;
}

/* Quick Access */
.quick-access-list {
  display: flex;
  flex-direction: column;
  gap: 15px;
}

.quick-access-item {
  display: flex;
  align-items: center;
  gap: 12px;
  padding: 12px;
  border-radius: 8px;
  cursor: pointer;
  transition: background-color 0.3s;
  border-left: 3px solid #b00d15;
}

.quick-access-item:hover {
  background-color: #f8f9fa;
}

.access-icon {
  width: 35px;
  height: 35px;
  border-radius: 50%;
  background-color: #f8f9fa;
  display: flex;
  align-items: center;
  justify-content: center;
  color: #b00d15;
}

.quick-access-item span {
  font-size: 0.95rem;
  color: #333;
  font-weight: 500;
}

/* Responsive Design */
@media (max-width: 1024px) {
  .dashboard-content {
    flex-direction: column;
    padding: 20px;
  }

  .sidebar {
    width: 100%;
  }

  .dashboard-grid {
    grid-template-columns: 1fr;
  }

  .stats-grid {
    grid-template-columns: 1fr;
  }
}

@media (max-width: 768px) {
  .dashboard-header {
    padding: 15px 20px;
  }

  .dashboard-header h1 {
    font-size: 1.5rem;
  }

  .profile-banner {
    flex-direction: column;
    gap: 15px;
    padding: 15px 20px;
    text-align: center;
  }

  .dashboard-content {
    padding: 15px;
  }

  .dashboard-card {
    padding: 20px;
  }
}

File: frontend/src/Dashboard/Management_dashboard.js
import "./Faculty_dashboard.css"
import React, { useEffect, useState } from 'react';
import ManagementProfile from '../Profile_details/Management_profile';

const ManagementDashboard = ({ firstName, lastName, onBack, onUpdateProfile, onCompleteProfile, onLogout }) => {
  const [staffCount, setStaffCount] = useState(0);
  const [studentCount, setStudentCount] = useState(0);

  useEffect(() => {
    fetch('http://localhost:8000/customers')
      .then((res) => res.json())
      .then((data) => {
        if (data && data.data) {
          const staff = data.data.filter((c) => c.customer_type === 'staff').length;
          const students = data.data.filter((c) => c.customer_type === 'student').length;
          setStaffCount(staff);
          setStudentCount(students);
        }
      })
      .catch((err) => {
        setStaffCount(0);
        setStudentCount(0);
      });
  }, []);

  return (
    <div className="dashboard-container">
      {/* Header */}
      <div className="header">
        <h1>Welcome, {firstName} {lastName}!</h1>
      </div>
      {/* Back Button Below Ribbon */}
      <div style={{ width: '100%', display: 'flex', justifyContent: 'flex-start', margin: '16px 0 0 32px' }}>
        <button className="management-back-btn" type="button" onClick={onBack} aria-label="Back">
          &#8592; Back
        </button>
      </div>

      {/* Profile Completion Banner */}
      <div className="profile-banner">
        <div className="profile-content">
          <div className="profile-text">
            <span>Your profile is filed is only 25%. Please update the remaining details.</span>
          <div className="progress-bar">
            <div className="progress-fill"></div>
          </div>
        </div>
        <button className="complete-profile-btn" onClick={onCompleteProfile}>Complete Profile</button>
        </div>
      </div>

      {/* Main Content */}
      <div className="main-content">
        {/* Management Dashboard Section */}
        <div className="dashboard-section">
          <h2 className="section-title">Management Dashboard</h2>

          {/* Main Cards */}
          <div className="main-cards">
            {/* Time Table Card */}
            <div className="card time-table-card">
                <div className="card-icon">
                <div className="icon-circle blue-bg">
                  <svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
                    <rect x="3" y="4" width="18" height="18" rx="2" ry="2" stroke="white" strokeWidth="2" />
                    <line x1="16" y1="2" x2="16" y2="6" stroke="white" strokeWidth="2" />
                    <line x1="8" y1="2" x2="8" y2="6" stroke="white" strokeWidth="2" />
                    <line x1="3" y1="10" x2="21" y2="10" stroke="white" strokeWidth="2" />
                  </svg>
                </div>
              </div>
                <h3>Time Table Configuration</h3>
              <p className="card-description">Manage class schedules</p>
              <p className="card-info">Next: Math at 10:00 AM</p>
              <button className="card-button">View Schedule</button>
              </div>

              {/* User Management Card */}
            <div className="card attendance-report-card">
                <div className="card-icon">
                <div className="icon-circle green-bg">
                  <svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
                    <path d="M16 21v-2a4 4 0 0 0-4-4H5a4 4 0 0 0-4 4v2" stroke="white" strokeWidth="2" />
                    <circle cx="8.5" cy="7" r="4" stroke="white" strokeWidth="2" />
                    <line x1="20" y1="8" x2="20" y2="14" stroke="white" strokeWidth="2" />
                    <line x1="23" y1="11" x2="17" y2="11" stroke="white" strokeWidth="2" />
                  </svg>
                </div>
              </div>
              <h3>User Management</h3>
              <p className="card-description">Manage user accounts</p>
              <p className="card-info">Current: 85% (17/20 users)</p>
              <button className="card-button">View Users</button>
            </div>

            {/* Faculty Count Card */}
            <div className="card faculty-count-card" style={{ background: '#f3f4f6', boxShadow: '0 2px 8px rgba(0,0,0,0.04)' }}>
              <div className="card-icon">
                <div className="icon-circle faculty-count-bg" style={{ backgroundColor: '#6366f1' }}>
                  <svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
                    <circle cx="12" cy="8" r="4" stroke="white" strokeWidth="2" />
                    <path d="M2 20c0-3.3137 3.134-6 7-6s7 2.6863 7 6" stroke="white" strokeWidth="2" />
                  </svg>
                </div>
              </div>
              <h3>Faculty Count</h3>
              <p className="card-description">Check total faculty</p>
              <p className="card-info">Available Faculty: {staffCount}</p>
            </div>

            {/* Student Count Card */}
            <div className="card students-count-card" style={{ background: '#f3f4f6', boxShadow: '0 2px 8px rgba(0,0,0,0.04)' }}>
              <div className="card-icon">
                <div className="icon-circle students-count-bg" style={{ backgroundColor: '#4F46E5' }}>
                  <svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
                    <path d="M16 21v-2a4 4 0 0 0-4-4H5a4 4 0 0 0-4 4v2" stroke="white" strokeWidth="2" />
                    <circle cx="8.5" cy="7" r="4" stroke="white" strokeWidth="2" />
                    <path d="M20 8v6" stroke="white" strokeWidth="2" />
                    <path d="M23 11h-6" stroke="white" strokeWidth="2" />
                  </svg>
                </div>
              </div>
              <h3>Students Count</h3>
              <p className="card-description">Check total students</p>
              <p className="card-info">Including all branches: {studentCount} </p>
            </div>
          </div>
        </div>

                {/* Quick Access Section */}
        <div className="quick-access-section">
          <h2 className="section-title">Quick Access</h2>
          <div className="quick-access-menu">
            <div className="menu-item" onClick={onUpdateProfile} style={{ cursor: 'pointer' }}>
              <div className="menu-icon">
                <svg width="16" height="16" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
                  <rect x="3" y="11" width="18" height="11" rx="2" ry="2" stroke="currentColor" strokeWidth="2" />
                  <circle cx="12" cy="16" r="1" fill="currentColor" />
                  <path d="M7 11V7a5 5 0 0 1 10 0v4" stroke="currentColor" strokeWidth="2" />
                </svg>
              </div>
              <span>Update Profile</span>
            </div>
            <div className="menu-item">
              <div className="menu-icon">
                <svg width="16" height="16" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
                  <circle cx="12" cy="12" r="3" stroke="currentColor" strokeWidth="2" />
                  <path d="M12 1v6m0 6v6m11-7h-6m-6 0H1" stroke="currentColor" strokeWidth="2" />
                </svg>
              </div>
              <span>Change Password</span>
            </div>
            <div className="menu-item" onClick={onLogout} style={{ cursor: 'pointer' }}>
              <div className="menu-icon">
                <svg width="16" height="16" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
                  <path d="M9 21H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h4" stroke="currentColor" strokeWidth="2" />
                  <polyline points="16,17 21,12 16,7" stroke="currentColor" strokeWidth="2" />
                  <line x1="21" y1="12" x2="9" y2="12" stroke="currentColor" strokeWidth="2" />
                </svg>
              </div>
              <span>Logout</span>
            </div>
          </div>
        </div>
      </div>
    </div>
  )
}

export default ManagementDashboard
